### PR TITLE
gitlab_runners inventory plugin: runtime en var and list default behavior

### PR DESCRIPTION
**Pull request by kalw**
Ability to pick options values from env vars on gitlab_runners inventory plugin
Remove default 20 items limit to runners list on gitlab_runners inventory plugin

##### SUMMARY

Remove default 20 items limit to runners list by adding "all" bool to True in gl.runners.all ref
Permit env var input for server_url and api_token options values

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

plugins/inventory/gitlab_runners.py
##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: plugins/inventory/gitlab_runners.py [hunk 1]
```diff
@@ -30,11 +30,15 @@
               - gitlab_runners
         server_url:
             description: The URL of the GitLab server, with protocol (i.e. http or https).
+            env:
+              - name: GITLAB_SERVER_URL
```
Comment (felixfontein) [inline plugins/inventory/gitlab_runners.py:34]:
```suggestion
              - name: GITLAB_SERVER_URL
                version_added: 1.0.0
```
Same for the other.

File: plugins/inventory/gitlab_runners.py [hunk 2]
```diff
@@ -95,7 +100,8 @@ def _populate(self):
             if self.get_option('filter'):
                 runners = gl.runners.all(scope=self.get_option('filter'))
             else:
-                runners = gl.runners.all()
+                runners = gl.runners.all(all=True)
+                version_added: 1.0.0
```
Comment (felixfontein) [inline plugins/inventory/gitlab_runners.py:104]:
```suggestion
```
It needs to be added for the `api_token` ini option, not randomly inside the code!

**Comment by felixfontein** [comment]
What you have been adding are not changelog fragments, but `version_added`s. You still need a changelog fragment. (Check the docs I linked for information on how they look like.)

**Comment by kalw** [comment]
> What you have been adding are not changelog fragments, but `version_added`s. You still need a changelog fragment. (Check the docs I linked for information on how they look like.)

👀

**Comment by kalw** [comment]
> What you have been adding are not changelog fragments, but `version_added`s. You still need a changelog fragment. (Check the docs I linked for information on how they look like.)

thanks for your patience, should be good :)

File: changelogs/fragments/611-gitlab-runners-env-vars-intput-and-default-item-limit.yaml [hunk 3]
```diff
@@ -0,0 +1,4 @@
+bugfixes:
+  - plugin inventory gitlab_runners - Remove default 20 items limit to runners list by adding "all" bool to True 
+  in gl.runners.all (https://github.com/python-gitlab/python-gitlab/blob/df485a92b713a0f2f983c72d9d41ea3a771abf88/gitlab/v4/objects.py#L4922)
+  Permit env var input for server_url and api_token options values
```
Comment (felixfontein) [inline changelogs/fragments/611-gitlab-runners-env-vars-intput-and-default-item-limit.yaml:4]:
```suggestion
  - gitlab_runners inventory plugin - make sure that all runners are returned, and not only the first 20 items (https://github.com/ansible-collections/community.general/pull/611).
minor_changes:
  - gitlab_runners inventory plugin - permit environment variable input for ``server_url`` and ``api_token`` options (https://github.com/ansible-collections/community.general/pull/611). values.
```

**Comment by felixfontein** [comment]
CC @Lunik @SamyCoenen @Shaps @dj-wasabi @marwatk @waheedi - you are maintainers for the gitlab_runner module, so maybe you also have an opinion (or can comment) on this PR for the gitlab_runners inventory plugin.

**Comment by felixfontein** [comment]
I just saw #440 - that is probably relevant here as well. Maybe the `all=True` change should be behind a switch?

**Comment by kalw** [comment]
> I just saw #440 - that is probably relevant here as well. Maybe the `all=True` change should be behind a switch?

sure, but at least warning this defaults to a subset of the object

File: changelogs/fragments/611-gitlab-runners-env-vars-intput-and-default-item-limit.yaml [hunk 4]
```diff
@@ -0,0 +1,4 @@
+bugfixes:
+  - gitlab_runners inventory plugin - make sure that all runners are returned, and not only the first 20 items (https://github.com/ansible-collections/community.general/pull/611).
+minor_changes:
+  - gitlab_runners inventory plugin - permit environment variable input for ``server_url`` and ``api_token`` options (https://github.com/ansible-collections/community.general/pull/611). values.
```
Comment (Andersson007) [inline changelogs/fragments/611-gitlab-runners-env-vars-intput-and-default-item-limit.yaml:4]:
```suggestion
  - gitlab_runners inventory plugin - permit environment variable input for ``server_url`` and ``api_token`` options (https://github.com/ansible-collections/community.general/pull/611).
```

**Comment by kalw** [comment]
> I just saw #440 - that is probably relevant here as well. Maybe the `all=True` change should be behind a switch?

removed default filter and added os env var capability for filter

File: changelogs/fragments/611-gitlab-runners-env-vars-intput-and-default-item-limit.yaml [hunk 5]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+  - gitlab_runners inventory plugin - permit environment variable input for ``server_url``, ``api_token`` and ``filter`` options (https://github.com/ansible-collections/community.general/pull/611).
```
Comment (Andersson007) [inline changelogs/fragments/611-gitlab-runners-env-vars-intput-and-default-item-limit.yaml:2]:
```suggestion
  - gitlab_runners - permit environment variable input for ``server_url``, ``api_token`` and ``filter`` options (https://github.com/ansible-collections/community.general/pull/611).
```
I fixed that example of fragment a couple of days ago.

**Comment by Andersson007** [comment]
I just emailed the initial author, let's wait for him for a day

**Comment by morph027** [comment]
Looks good to me.

File: changelogs/fragments/611-gitlab-runners-env-vars-intput-and-default-item-limit.yaml (hunk 5 above)
Comment (Andersson007) [reply changelogs/fragments/611-gitlab-runners-env-vars-intput-and-default-item-limit.yaml:2]: ok, there's no reason to wait

**Comment by Andersson007** [comment]
@kalw thanks for the contribution!
@felixfontein @morph027 thanks for reviewing!
Merged https://github.com/ansible-collections/community.general/pull/611 into main

**Pull request merged**